recipe-client-addon/lib/EventEmitter.jsm

```diff
@@ -1,34 +1,42 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-// This file is meant to run inside action sandboxes
-
 "use strict";
 
+const {utils: Cu} = Components;
+const {jsonCopy} = Cu.import("resource://shield-recipe-client/lib/Utils.jsm", {});
+Cu.import("resource://shield-recipe-client/lib/LogManager.jsm");
+
+this.EXPORTED_SYMBOLS = ["EventEmitter"];
 
-this.EventEmitter = function(driver) {
-  if (!driver) {
-    throw new Error("driver must be provided");
-  }
+const log = LogManager.getLogger("event-emitter");
 
+this.EventEmitter = function(sandboxManager) {
   const listeners = {};
 
   return {
+    createSandboxedEmitter() {
+      return sandboxManager.cloneInto({
```

Comment:
(I checked only the `EventEmitter` browser test, and that doesn't seem to make calls from inside the sandbox...)